src/engine/SpatialJoinAlgorithms.cpp

```diff
@@ -37,19 +38,103 @@ std::optional<GeoPoint> SpatialJoinAlgorithms::getPoint(const IdTable* restable,
 Id SpatialJoinAlgorithms::computeDist(const IdTable* idTableLeft,
                                       const IdTable* idTableRight,
                                       size_t rowLeft, size_t rowRight,
                                       ColumnIndex leftPointCol,
                                       ColumnIndex rightPointCol) const {
+  auto getAreaString = [&](const IdTable* idtable, size_t row, size_t col) {
+    return betweenQuotes(ExportQueryExecutionTrees::idToStringAndType(
+                             qec_->getIndex(), idtable->at(row, col), {})
+                             .value()
+                             .first);
+  };
+
+  auto getAreaOrPointGeometry =
+      [&](const IdTable* idtable, size_t row, size_t col,
+          std::optional<GeoPoint> point) -> std::optional<AnyGeometry> {
+    AnyGeometry geometry;
+    try {
+      if (!point) {
+        boost::geometry::read_wkt(getAreaString(idtable, row, col), geometry);
+      } else {
+        geometry = Point(point.value().getLng(), point.value().getLat());
+      }
+    } catch (...) {
+      return std::nullopt;
+    }
+    return geometry;
+  };
+
+  // for now we need to get the data from the disk, but in the future, this
+  // information will be stored in an ID, just like GeoPoint
+  auto getAreaPoint = [&](const IdTable* idtable, size_t row,
+                          size_t col) -> std::optional<GeoPoint> {
+    std::string areastring = getAreaString(idtable, row, col);
+
+    Box areaBox;
+    try {
+      areaBox = calculateBoundingBoxOfArea(areastring);
+    } catch (...) {
+      // nothing to do. When parsing a point or an area fails, a warning message
+      // gets printed at another place and the point/area just gets skipped
+      return std::nullopt;
+    }
+    Point p = calculateMidpointOfBox(areaBox);
+    return GeoPoint(p.get<1>(), p.get<0>());
+  };
+
   auto point1 = getPoint(idTableLeft, rowLeft, leftPointCol);
   auto point2 = getPoint(idTableRight, rowRight, rightPointCol);
-  if (!point1.has_value() || !point2.has_value()) {
-    return Id::makeUndefined();
+  if (useMidpointForAreas_) {
+    if (!point1) {
+      point1 = getAreaPoint(idTableLeft, rowLeft, leftPointCol);
+    }
+
+    if (!point2) {
+      point2 = getAreaPoint(idTableRight, rowRight, rightPointCol);
+    }
+
+    if (!point1.has_value() || !point2.has_value()) {
+      return Id::makeUndefined();
+    }
+    return Id::makeFromDouble(
+        ad_utility::detail::wktDistImpl(point1.value(), point2.value()));
+  } else {
+    auto geometry1 =
+        getAreaOrPointGeometry(idTableLeft, rowLeft, leftPointCol, point1);
+    auto geometry2 =
+        getAreaOrPointGeometry(idTableRight, rowRight, rightPointCol, point2);
+    if (!geometry1.has_value() || !geometry2.has_value()) {
+      return Id::makeUndefined();
+    }
+    // convert to m and return. Note that the 78630 is an approximation, which
+    // does not provide accurate results for the poles.
```

Comment:
What exactly is `78630`?
Can you link to some doku/reference/the name of this thing?